numpy/lib/nanfunctions.py

```diff
@@ -1363,19 +1363,23 @@ def _nanquantile_unchecked(a, q, axis=None, out=None, overwrite_input=False,
     # apply_along_axis in _nanpercentile doesn't handle empty arrays well,
     # so deal them upfront
     if a.size == 0:
-        return np.nanmean(a, axis, out=out, keepdims=keepdims)
-
-    r, k = function_base._ureduce(
-        a, func=_nanquantile_ureduce_func, q=q, axis=axis, out=out,
-        overwrite_input=overwrite_input, interpolation=interpolation
-    )
-    if keepdims and keepdims is not np._NoValue:
-        return r.reshape(q.shape + k)
+        r = np.nanmean(a, axis, keepdims=keepdims)
+        if q.ndim != 0:
+            r = np.broadcast_to(r, q.shape + r.shape)
```

Comment:
I'm thinking the actual solution may be to remove this branch entirely, and re-insert it in `_nanquantile_ureduce_func` - that way the broadcasting should be automatically handled by `_ureduce`